feat(tools): add action to check for translations

naomi-lgbt opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `master` branch of freeCodeCamp.
- [X] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

<!-- Feel free to add any additional description of changes below this line -->

This adds a very basic GitHub action to help catch PRs that modify translated curriculum challenges. Here's the logic:

1. A PR changes only English challenges (files within `curriculum/challenges/english`):
  - The workflow will *not* run.
2. A PR changes non-English challenges (files within `curriculum/challenges/!english`):
  - The workflow will run and will throw an error to fail.
3. The Crowdin integration opens a PR to update the translated files:
  - The workflow will run, but because the PR author is `github-actions` the action will not hit the error block and will pass.

I created a sample repository with three PRs to demonstrate these three scenarios:
https://github.com/nhcarrigan/actions-testing/pulls

@RandellDawson would you mind giving this a look?

Commit:
feat(tools): add action to check for translations

Adds a GitHub action that checks if translated curriculum challenges
have been modified. Will directly ignore the branches created by
Crowdin.

Signed-off-by: nhcarrigan <nhcarrigan@gmail.com>

Commit:
fix: Check by username

Signed-off-by: nhcarrigan <nhcarrigan@gmail.com>

RandellDawson commented on .github/workflows/no-prs-to-translation.yml:
```diff
@@ -0,0 +1,18 @@
+      if: github.event.pull_request.user.login != "github-actions[bot]"
+      with:
+        script: |
+          throw "You appear to be translating files directly on GitHub. Please use Crowdin instead."
```
```suggestion
          throw "You appear to be translating files directly on GitHub. Please visit https://translate.freecodecamp.org instead."
```

Commit:
Update .github/workflows/no-prs-to-translation.yml

Co-authored-by: Randell Dawson <5313213+RandellDawson@users.noreply.github.com>

Commit:
add comment feature

Signed-off-by: nhcarrigan <nhcarrigan@gmail.com>

Commit:
fix: use pull_request_target

Signed-off-by: nhcarrigan <nhcarrigan@gmail.com>

RandellDawson reviewed (dismissed):
I tested this action on @nhcarrigan's personal repo by creating various PRs to get the action to fire and the action successfully sets the status to fail and makes the applicable comment on the PR even from a forked repo.

I am fine with these changes.  This would complete the following task outlined under **Still To Do** on PR #40657:

> Need to create a GitHub action that does not permit a PR to be merged to master if it includes any files in the non-English curriculum folders. This will make sure all changes from Crowdin are the source for those folders.

RandellDawson commented:
@nhcarrigan I just validated the action catches files that were deleted too.  I am now thinking the message/comment should be something like:
> You appear to be adding/modifying/deleting non-English curriculum files directly on GitHub. If you are trying to make translation changes to a curriculum challenge file, please visit https://translate.freecodecamp.org instead.

RandellDawson commented:
We need to hold off merging this PR until all the existing Chinese curriculum files have been merged.

Commit:
tools: update comment text

Updates the text the action uses when leaving a PR comment for apparent
translation PRs.

Signed-off-by: nhcarrigan <nhcarrigan@gmail.com>
